fmn/fmn_fasshim.py

```diff
@@ -57,79 +74,96 @@ def make_fas_cache(**config):
-            auth=True)
-    except fedora.client.ServerError as e:
-        log.warning(
-            "Failed to download fas cache for %s: %r" % (username, e))
-        return {}
-
-    log.info("Caching necessary data for %s" % username)
-    for user in request['people']:
-        nick = user['ircnick']
-        if nick:
-            _cache.set(nick, user['username'])
-
-        email = user['email']
-        if email:
-            _cache.set(email, user['username'])
+    global client
+    if config.get('fasjson'):
+        try:
+            log.info("Downloading FASJSON cache for %s*" % username)
+            response = client.get_user(username=username)
+            _add_to_cache([response.result])
+        except fasjson_client.errors.APIError as e:
+            log.error("Something went wrong updating the cache with error: %s" % e)
     else:
-        # If we couldn't find the nick in FAS, save it in the _cache as nick
-        # so that we avoid calling FAS for every single filter we have to
-        # run through
-        _cache.set(username, username)
+        try:
+            log.info("Downloading FAS cache for %s*" % username)
+            request = client.send_request(
+                '/user/list',
+                req_params={'search': '%s' % username},
+                auth=True)
+        except fedora.client.ServerError as e:
+            log.warning(
+                "Failed to download fas cache for %s: %r" % (username, e))
+            return {}
+
+        log.info("Caching necessary data for %s" % username)
+        for user in request['people']:
+            nick = user['ircnick']
+            if nick:
+                _cache.set(nick, user['username'])
+
+            email = user['email']
+            if email:
+                _cache.set(email, user['username'])
+        else:
+            # If we couldn't find the nick in FAS, save it in the _cache as nick
+            # so that we avoid calling FAS for every single filter we have to
+            # run through
+            _cache.set(username, username)
 
 
 def update_email(email):
-    global fasclient
-    try:
-        log.info("Downloading FAS cache for %s" % email)
-        request = fasclient.send_request(
-            '/user/list',
-            req_params={
-                'search': '%s' % email,
-                'by_email': 1,
-            },
-            auth=True)
-    except fedora.client.ServerError as e:
-        log.warning(
-            "Failed to download fas cache for %s: %r" % (email, e))
-        return {}
-
-    log.info("Caching necessary data for %s" % email)
-    for user in request['people']:
-        nick = user['ircnick']
-        if nick:
-            _cache.set(nick, user['username'])
-
-        email = user['email']
-        if email:
-            _cache.set(email, user['username'])
+    global client
+    if config.get('fasjson'):
+        try:
+            log.info("Downloading FASJSON cache for %s*" % username)
```

Comment:
Flake8 caught that this should be `email` because `username` is undefined. Flake8 is nice ;-)